### PR TITLE
migrate to RestExchange

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/exchanges/adapters/abstract_adapter.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        try:
+            adapted = func(*args, **kwargs)
+            # add any other common adapter function logic here
+            return adapted
+        except Exception as err:
+            raise errors.UnexpectedAdapterError(f"Unexpected error when adapting exchange data: {err}") from err
+    return wrapper
+
+
+class AbstractAdapter:
```
Comment (Herklos) [inline octobot_trading/exchanges/adapters/abstract_adapter.py:34]: 💯

File: octobot_trading/api/exchange.py [hunk 2]
```diff
@@ -43,12 +44,13 @@ def get_exchange_manager_from_exchange_name_and_id(exchange_name: str, exchange_
 
 async def get_exchange_available_time_frames(
```
Comment (Herklos) [inline octobot_trading/api/exchange.py:45]: Should we name it `get_ccxt_exchange_available_time_frames` as it is only able to fetch ccxt exchange time frames ?

File: octobot_trading/exchange_data/recent_trades/channel/recent_trade_updater.py [hunk 3]
```diff
@@ -37,9 +37,7 @@ async def init_recent_trades(self):
                 recent_trades = await self.channel.exchange_manager.exchange.\
                     get_recent_trades(pair, limit=self.RECENT_TRADE_LIMIT)
                 if recent_trades:
-                    await self.push(pair,
-                                    list(map(self.channel.exchange_manager.exchange.clean_recent_trade,
-                                             recent_trades)))
+                    await self.push(pair, recent_trades)
```
Comment (Herklos) [inline octobot_trading/exchange_data/recent_trades/channel/recent_trade_updater.py:40]: :+1:

File: octobot_trading/exchanges/abstract_exchange.py [hunk 4]
```diff
@@ -520,34 +520,6 @@ def parse_account(self, account):
         """
         raise NotImplementedError("parse_account is not implemented")
 
-    """
-    Cleaners
```
Comment (Herklos) [inline octobot_trading/exchanges/abstract_exchange.py:524]: :+1:

File: octobot_trading/exchanges/adapters/abstract_adapter.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import decimal
+
+import octobot_trading.errors as errors
+import octobot_commons.logging as logging
+import octobot_commons.constants as commons_constants
+
+
+def _adapter(func):
```
Comment (Herklos) [inline octobot_trading/exchanges/adapters/abstract_adapter.py:23]: :heart_eyes:

File: octobot_trading/exchanges/adapters/abstract_adapter.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        return self.parse_mark_price(fixed, **kwargs)
+
+    @_adapter
+    def adapt_market_status(self, raw, **kwargs):
+        fixed = self.fix_market_status(raw, **kwargs)
+        return self.parse_market_status(fixed, **kwargs)
+
+    def get_uniformized_timestamp(self, timestamp):
+        # override if the exchange time is not a second timestamp or millisecond
+        if timestamp > 16728292300:  # Friday 5 February 2500 11:51:40
```
Comment (Herklos) [inline octobot_trading/exchanges/adapters/abstract_adapter.py:101]: :smile:

File: octobot_trading/exchanges/types/rest_exchange.py [hunk 7]
```diff
@@ -18,38 +17,60 @@
...
-#TODO remove
-class SpotCCXTExchange(exchanges_types.SpotExchange):
+
+class RestExchange(abstract_exchange.AbstractExchange):
     ORDER_NON_EMPTY_FIELDS = [ecoc.ID.value, ecoc.TIMESTAMP.value, ecoc.SYMBOL.value, ecoc.TYPE.value,
                               ecoc.SIDE.value, ecoc.PRICE.value, ecoc.AMOUNT.value, ecoc.STATUS.value]
     ORDER_REQUIRED_FIELDS = ORDER_NON_EMPTY_FIELDS + [ecoc.REMAINING.value]
-    CONNECTOR_CLASS = exchange_connectors.CCXTExchange
+    """
+    RestExchange is using its exchange connector to interact with the exchange.
```
Comment (Herklos) [inline octobot_trading/exchanges/types/rest_exchange.py:43]: :+1:

**Review by Herklos — APPROVED** [review]
Great refactor work :100:

File: octobot_trading/api/exchange.py (hunk 2 above)
Comment (GuillaumeDSM) [reply octobot_trading/api/exchange.py:45]: right

**Review by Herklos — APPROVED** [review]
👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**